syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -411,17 +398,35 @@ slng_passwd(int argc, char *argv[], const gchar *mode)
   gint buff_size = 255;
   gchar buff[buff_size+1];
 
-  if (g_strcmp0(argv[1],"store")==0 && (argv[3]==NULL || strlen(argv[3])==0))
+  if (g_strcmp0(mode,"password-add")==0)
     {
-      gulong password_length = buff_size;
-      gchar passwd[password_length+1];
-      get_password_from_stdin(passwd,&password_length);
-      snprintf(buff, buff_size, "PWD %s %s %s", argv[1], argv[2], passwd);
-      memset(passwd,'?',password_length);
+      if (argv[1]==NULL || strlen(argv[1])==0)
+        {
+          print_usage(argv[0]);
+          return 1;
+        }
+
+      if (argv[2]==NULL || strlen(argv[2])==0)
+        {
+          gulong password_length = buff_size;
+          gchar password_buffer[password_length+1];
+          get_password_from_stdin(password_buffer,&password_length);
+          snprintf(buff, buff_size, "PWD %s %s %s", "add", argv[1], password_buffer);
+          memset(password_buffer,'?',password_length);
+        }
+      else
+        {
+          snprintf(buff, buff_size, "PWD %s %s %s", "add", argv[1], argv[2]);
```

Comment:
Lets use glib versions whenever possible: g_snprintf

Reply:
replaced all snprintf to g_snprintf